linked_list/main.py

```diff
@@ -42,54 +41,190 @@ class Node(object):
+        if not isinstance(index, int):
+            err_msg = '"index" must be an integer ({} given)'
+            raise TypeError(err_msg.format(type(index)))
+        
+        # Iterate through the items and return the one with the right index
+        node_index = 0
+        for node in self:
+            if index == node_index:
+                return node
+            node_index += 1
+        
+        # Raise an IndexError if no item matches the index supplied
+        err_msg = 'LinkedList index out of range'
+        raise IndexError(err_msg)
+    
     def __add__(self, other):
-        pass
-
+        # Raise a TypeError if anything other than a LinkedList is added
+        if not isinstance(other, LinkedList):
+            err_msg = 'Only another LinkedList object may be added \
+                to a LinkedList'
+            raise TypeError(err_msg)
+        
+        # Build a new list from scratch
+        new_list = []
+        for node in self:
+            new_list.append(node.elem)
+        for node in other:
+            new_list.append(node.elem)
+        
+        # Return the LinkedList version of it
+        return LinkedList(new_list)
+    
     def __iadd__(self, other):
-        pass
-
+        # Raise a TypeError if anything other than a LinkedList is added
+        if not isinstance(other, LinkedList):
+            err_msg = 'Only another LinkedList object may be added \
+                to a LinkedList'
+            raise TypeError(err_msg)
+        
+        # The nodes list prevents the LinkedList from growing infinitely large 
+        #  if it is added to itself
+        nodes = [node for node in other]
+        for node in nodes:
+            self.append(node.elem)
+        
+        return self
+    
     def __eq__(self, other):
-        pass
-
+        # If other is not a LinkedList, they are not equal
+        if not isinstance(other, LinkedList):
+            return False
+        
+        # If the lists have unequal length, they are not equal
+        if len(self) != len(other):
+            return False
+        
+        # If there are any nodes in the list whose elements are different,
+        #  they are not equal
+        for self_node, other_node in zip(self, other):
+            if self_node.elem != other_node.elem:
+                return False
+        
+        # If the lists have equal length, and every node is the same, they are
+        #  equal
+        return True
+    
     def append(self, elem):
-        pass
-
+        # Make a node of the element to be appended
+        this = Node(elem)
+        
+        # Make this node "next" for the previously last node
+        if self.end:
+            self.end.next = this
+        else:
+            self.start = this
+        
+        # Make this the end element
+        self.end = this
+    
     def count(self):
-        pass
-
+        return len(self)
+    
     def pop(self, index=None):
```

Comment:
I wasn't assign to review this code but just wanted to say it looks great and I learned a lot read through it!

The only possible issue I saw with not setting self.end to None when popping from a list with just 1 element
    l = LinkedList([1])
    l.pop()
    l.append(5)
    print l # l will be an empty list

But overall awesome code!